ldm/modules/textual_inversion_manager.py

```diff
@@ -89,27 +92,45 @@ def load_textual_inversion(self, ckpt_path: Union[str,Path], defer_injecting_tok
             return
         elif (
             self.text_encoder.get_input_embeddings().weight.data[0].shape[0]
-            != embedding_info['token_dim']
+            != embedding_info["token_dim"]
         ):
             print(
                 f"** Notice: {ckpt_path.parents[0].name}/{ckpt_path.name} was trained on a model with an incompatible token dimension: {self.text_encoder.get_input_embeddings().weight.data[0].shape[0]} vs {embedding_info['token_dim']}."
             )
             return
 
-        if embedding_info:
-            try:
-                self._add_textual_inversion(
-                    embedding_info["name"],
-                    embedding_info["embedding"],
-                    defer_injecting_tokens=defer_injecting_tokens,
-                )
-            except ValueError as e:
-                print(f'   | Ignoring incompatible embedding {embedding_info["name"]}')
-                print(f"   | The error was {str(e)}")
-        else:
+        # Resolve the situation in which an earlier embedding has claimed the same
+        # trigger string. We replace the trigger with '<source_file>', as we used to.
+        trigger_str = embedding_info["name"]
+        sourcefile = (
+            f"{ckpt_path.parent.name}/{ckpt_path.name}"
+            if ckpt_path.name == "learned_embeds.bin"
+            else ckpt_path.name
+        )
+
+        if trigger_str in self.trigger_to_sourcefile:
+            replacement_trigger_str = (
+                f"<{ckpt_path.parent.name}>"
+                if ckpt_path.name == "learned_embeds.bin"
+                else f"<{ckpt_path.stem}>"
+            )
             print(
-                f">> Failed to load embedding located at {str(ckpt_path)}. Unsupported file."
+                f">> {sourcefile}: Trigger token '{trigger_str}' is already claimed by '{self.trigger_to_sourcefile[trigger_str]}'. Trigger this concept with {replacement_trigger_str}"
             )
+            trigger_str = replacement_trigger_str
+
+        try:
+            self._add_textual_inversion(
+                trigger_str,
+                embedding_info["embedding"],
+                defer_injecting_tokens=defer_injecting_tokens,
+            )
+        except ValueError as e:
+            print(f'   | Ignoring incompatible embedding {embedding_info["name"]}')
+            print(f"   | The error was {str(e)}")
+
+        # remember which source file claims this trigger
+        self.trigger_to_sourcefile[trigger_str] = sourcefile
```

Comment:
Shouldn't this only be done in the `try` block above? Otherwise we shouldn't "reserve" the trigger string.

Reply:
Everything else LGTM.

Reply:
Right you are! All done.